### ISSUE TITLE
Components automatically injected into the root element

### ISSUE THREAD

**Issue by imlautaro** [nuxt/nuxt#19492]
### Describe the feature

Sometimes you need to have multiple components in the `app.vue` file, or as close to the root element as possible. For example, if you have multiple modals that must be present not only on every page, but also on every layout (dialogs, cookie notice, etc) or any other interface resource and you must add each one manually somewhere in the `app.vue` file.

I think it would be nice to have a directory for "universal" components that Nuxt detects and automatically injects into the root element. This would perfectly follow the philosophy of the framework to improve the DX as much as possible.

An example of how it would be now and how it could be:

```vue
// ~/app.vue
<template>
  <div>
    <NuxtLayout>
      <NuxtPage />
    </NuxtLayout>
    <!-- universal components -->
    <MenuDialog />
    <LanguageDialog />
    <ColorModeDialog />
    <Backdrop />
    <!-- etc -->
  </div>
</template>
```
Or you could just move those components to a directory called `~/components/universal` (for example) and Nuxt would take care of injecting them automatically.

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by AloisSeckar** [comment]
I would find a good use for that in my project, but I have a couple of questions/concerns/remarks:

1. Is that even a good pattern to have a bunch of components, mostly modal windows I guess, like that? I mean, I came up with the very same solution to handle modals in my apps, but somehow I dont like it, especially as the number of such components tends to grow as the project grows... 
2. If yes, they can still quite easilly be wrapped in a dedicated component and the problem is gone. Or?
3. It adds a new level of complexity, because you should be aware that this new special directory exits and behaves differently than "normal" `components`. So are the potential benefits really worth it?
4. If my previous point is correct - ain't that rather a use case for a new module or plugin? Because some people may benefit from it, while others won't. 

Please don't take it like I want to mock or destroy your idea. This is just what came up my mind right now.

**Comment by imlautaro** [comment]
First of all, thanks for the feedback! I'll tell you what I think about each point you mention:

1. This feature in Nuxt by default would precisely allow the code to follow a more logical structure and a better pattern, because at the end of the day in relatively large projects you will always need UI elements that are available in every route and layout, so instead of having to manually add them to the `app.vue` file where those components have no relationship with each other and do not share a state, it would make more sense for them to be in a universal folder and for Nuxt to inject them in the background.
2. Yes, you can safely place each component inside a dedicated one, but we would have the same "hassle" of having to add a bunch of unrelated components to the same file just so they can be in the app.
3. Regarding the complexity, I don't think it is harmful in the least, I think it would be somewhat intuitive to have these components in a dedicated folder as well as we have the global components and in the case of nuxt content the components to use in the MDC . It could also be seen as what we already have with middlewares, where we can add the .global extension so they are automatically injected into each route. It would be pretty much the same applied to components, only in this case I think a directory makes more sense than using a .univeral.vue extension for example.
4. Finally, regarding this point I think it could be implemented as a feature within the nuxt package and if someone doesn't need this functionality, they simply don't use it and that doesn't interfere or overload anything else.

In short, I think it would be an intuitive feature that could improve the development experience for those who need it without annoying those who don't. So, let me know what you think and let's await for the Nuxt team to tell us what they think as well 😄
